handler/core/src/main/java/io/knotx/fragments/task/TaskManager.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * The code comes from https://github.com/tomaszmichalak/vertx-rx-map-reduce.
+ */
+package io.knotx.fragments.task;
+
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.engine.FragmentEventContext;
+import io.knotx.fragments.engine.Task;
+import io.knotx.fragments.handler.action.ActionProvider;
+import io.knotx.fragments.task.exception.GraphConfigurationException;
+import io.knotx.fragments.task.exception.TaskNotFoundException;
+import io.knotx.fragments.task.options.TaskOptions;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+
+public class TaskManager {
```

Comment:
That's not a manager
`TaskManager` here is an Abstract Factory while `TaskProviders` are concrete factories.

Reply:
Sure, something like this.

However, task manger:
- gets task name from fragment configuration
- find task in tasks map
- prepare configuration for task provider
- delegate task creation to task provider

So it is not factory, it keeps Handler's configs (like the map of tasks, a task configuration key), while tasks providers know about specific task.